### PR TITLE
[Utils] Revisit RealPath

### PR DESCRIPTION
**Pull request by dayo09**
This commit rewrites RealPath's equal commands.

ONE-vscode-DCO-1.0-Signed-off-by: dayo09 <dayoung.lee@samsung.com>

---

From #1011

### CODE REVIEW

File: src/Utils/Helpers.ts [hunk 1]
```diff
@@ -36,15 +36,35 @@ export class RealPath {
     this.absPath = absPath;
   }
 
-  public static isEqual(path0: RealPath, path1: RealPath): boolean {
-    return path0.absPath === path1.absPath;
+  public static areEqual(path0: string, path1: string): boolean {
```
Comment (dayo09) [inline src/Utils/Helpers.ts:39]: This function is handy to process `equal` to string paths.

File: src/Utils/Helpers.ts [hunk 2]
```diff
@@ -36,15 +36,35 @@ export class RealPath {
...
+    return this.absPath === lhs.absPath;
   }
 
   public static createRealPath(rawPath: string): RealPath|null {
     const absPath = path.resolve(path.normalize(rawPath));
 
     return fs.existsSync(absPath) ? new RealPath(absPath) : null;
   }
+
+  public static exist(rawPath: string|undefined): boolean {
```
Comment (dayo09) [inline src/Utils/Helpers.ts:60]: This command makes users easy to check if the path exists.

File: src/Utils/Helpers.ts (hunk 1 above)
Comment (seanshpark) [reply src/Utils/Helpers.ts:39]: I usually seen `is~~~` as `path0 is equal to path1 ?`
I've never seen `are~~~` and at first seeing this, I interpreted this as `path0: string[] are equal to path1: string[] ?`

File: src/Utils/Helpers.ts (hunk 2 above)
Comment (yunjayh) [reply src/Utils/Helpers.ts:60]: (optional) For me, `pathlib.Path.exists()` is friendly. How about `exists`?

File: src/Utils/Helpers.ts (hunk 1 above)
Comment (yunjayh) [reply src/Utils/Helpers.ts:39]: +1 to `isEqual`.

File: src/Utils/Helpers.ts [hunk 3]
```diff
@@ -36,15 +36,35 @@ export class RealPath {
...
+    if (!rawPath) {
+      return false;
+    }
```
Comment (yunjayh) [inline src/Utils/Helpers.ts:63]: IMHO for the case of undefined rawPath, how about returning the `this.abspath`'s existence? When I read this code, I was expected to use `exist` method with a `RealPath` instance, for example 
```ts
filepath = RealPath('~/my_file.cfg');
filepath.exist()
```

File: src/Utils/Helpers.ts (hunk 1 above)
Comment (dayo09) [reply src/Utils/Helpers.ts:39]: I tried to search on a google:

Unity assertion, .NET assertion, C# Selenium, etc,... Uses areEqual. (Google search result : 174,000,000+)
iBatis, Apple uses isEqual, (Apple's for an unary compare operator.) (Google search result: 827,000+)

File: src/Utils/Helpers.ts (hunk 3 above)
Comment (dayo09) [reply src/Utils/Helpers.ts:63]: It's not a member function, but a static function.

Usage: 
```
RealPath.exist('~/my_file.cfg');
```

Comment (yunjayh) [reply src/Utils/Helpers.ts:63]: 😮 Now I catch it..!

Comment (yunjayh) [reply src/Utils/Helpers.ts:63]: Now I catch it..! 😮

File: src/Utils/Helpers.ts (hunk 1 above)
Comment (dayo09) [reply src/Utils/Helpers.ts:39]: > I've never seen are~~~ and at first seeing this, I interpreted this as path0: string[] are equal to path1: string[] ?

As far as I know, code must be read as it is. In this example,

```areEqual(path0, path1)```

So people will read as: "Are equal: path0 and path1". It sounds gramatically correct.

Whereas this sounds awkward: "Is equal: path0 and path1". 

If it's an unary operator, it isEqual will sound more naturally:
```path0.isEqual(path1)```
"Path0 isEqual to path1"

Comment (seanshpark) [reply src/Utils/Helpers.ts:39]: This is your change so you can go as you wish.
But finding examples from internet with `areEqual`, with .NET, I'm not sure this is appropriate.
I've found some examples with TypeScript which I agree upon.
- https://dev.to/macsikora/typescript-is-more-than-you-think-2nbf
- https://www.anycodings.com/questions/how-to-compare-arrays-in-javascript

Anyway, it's up to you.

**Review by yunjayh — APPROVED** [review]
LGTM

**Comment by dayo09** [comment]
@Samsung/one-vscode PTAL :-D

**Review by YongseopKim — APPROVED** [review]
LGTM

**Pull request merged**